src/analysis/asymptotics/superpolynomial_decay.lean

```diff
@@ -0,0 +1,244 @@
+/-
+Copyright (c) 2021 Devon Tuma. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Devon Tuma
+-/
+import analysis.asymptotics.asymptotics
+import analysis.asymptotics.specific_asymptotics
+import data.polynomial.eval
+
+/-!
+# Super-Polynomial Function Decay
+
+This file defines a predicate `asymptotics.superpolynomial_decay f` for a function satisfying
+  one of following equivalent definitions (The definition is in terms of the first condition):
+
+* `f` is `O(x ^ c)` for all (or equivalently sufficiently small) integers `c`
+* `f` is `O(p(x)⁻¹)` for all (or equivalently sufficiently large) polynomials `p`
+* `p(x) * f` is bounded for all polynomials `p`
+* `p(x) * f` tends to `𝓝 0` for all polynomials `p`
+
+The definition of superpolynomial decay for a function `f : α → 𝕜`
+  is made relative to an algebra structure `[algebra α 𝕜]`.
+Negligibility then means the function `f x` decays faster than `(p.eval (algebra_map α 𝕜 x))⁻¹`
+  for all polynomials `p : polynomial 𝕜`.
+-/
+
```

Comment:
Can you also add a reference to something relevant in the literature? (Or just the wikipedia page.)